timemachine/fe/single_topology_v3.py

```diff
@@ -779,34 +995,71 @@ def setup_intermediate_state(self, lamb):
         """
         src_system = self.src_system
         dst_system = self.dst_system
-        interpolate_fn = interpolate.linear_interpolation
 
-        # tbd: use different interpolation functions later
+        # alchemical parameter values at which angle and torsion force constants transition from zero to nonzero for
+        # ring closing transformations
+        lambda_angles = 0.4  # for angles, src_k = 0 => k(lamb) = 0 when lamb < lambda_angles
+        lambda_torsions = 0.7  # analogous for torsions
+
         bond = self._setup_intermediate_bonded_term(
-            src_system.bond, dst_system.bond, lamb, interpolate.align_harmonic_bond_idxs_and_params, interpolate_fn
+            src_system.bond,
+            dst_system.bond,
+            lamb,
+            interpolate.align_harmonic_bond_idxs_and_params,
+            partial(
+                interpolate_harmonic_bond_params,
+                k_min=0.1,  # ~ 5 nm at 300 K
+                lambda_min=0.0,
+                lambda_max=lambda_angles,
+            ),
         )
+
         angle = self._setup_intermediate_bonded_term(
-            src_system.angle, dst_system.angle, lamb, interpolate.align_harmonic_angle_idxs_and_params, interpolate_fn
+            src_system.angle,
+            dst_system.angle,
+            lamb,
+            interpolate.align_harmonic_angle_idxs_and_params,
+            partial(
+                interpolate_harmonic_angle_params,
+                k_min=0.05,
```

Comment:
Optional nit: Add line comment (similar to the one for the value of `k_min` for harmonic_bonds above) if this value has been tuned / if there is a known acceptable range for this parameter?

Reply:
Addressed in [3003770](https://github.com/proteneer/timemachine/pull/820/commits/300377099ccf6981338d5c196bc508fc3289f815)